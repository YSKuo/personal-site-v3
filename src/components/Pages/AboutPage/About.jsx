import React from "react";
import styled from "styled-components";
import { Container, Grid, Button, IconButton } from "@material-ui/core";
import {
  Section,
  SectionTitle,
  SectionSubtitle,
  P,
  B,
  A,
} from "../../Section/Section";
import { mediaQueryBreakpoint } from "../../../constants/breakpoint";
import StyledDivider from "../../StyledDivider";

function About({ config }) {
  const {
    userName,
    userEmail,
    userResume,
    userLocation,
    userLinks,
    userSkills,
  } = config;

  const findUserLinks = (target) => {
    return userLinks.find(
      (item) => item.label.toLowerCase() === target.toLowerCase()
    );
  };

  const linkedinInfo = findUserLinks("linkedin");
  const instagramInfo = findUserLinks("instagram");

  return (
    <Container>
      <Section>
        <SectionTitle>About Me</SectionTitle>
        <P>
          Hi, there! I'm <B>{userName}</B>, a frontend developer and designer
          living in {userLocation}. After finishing my master's degree in
          mechanical engineering, I acquired extensive experience in a patent
          firm and also the high-tech manufacturing industry.
        </P>
        <P>
          Gaining a glimpse of user-centered methodologies from my experience in
          the manufacturing industry, I am eager to encourage interactions among
          people through digital products. Therefore, I've worked hard to
          improve my coding and design skills, which allow me to create websites
          that are intuitive and aesthetically balanced.
        </P>
        <P>
          In my spare time, I like to indulge myself in <B>music</B>, broaden my
          horizons with <B>travel</B>, document my journey by <B>photographs</B>
          , cultivate my taste in <B>denim</B> and <B>heritage menswear</B>, and
          feed my imagination of impossibilities with <B>video games</B> and{" "}
          <B>films</B>.
        </P>
        <P>
          You may also know more about me through my{" "}
          <B>
            <A href={linkedinInfo.url}>{linkedinInfo.label}</A>
          </B>{" "}
          or{" "}
          <B>
            {" "}
            <A href={instagramInfo.url}>{instagramInfo.label}</A>
          </B>
          .
        </P>
      </Section>
      <StyledDivider yMargin="2rem" />
      <Section>
        <SectionTitle>Skills</SectionTitle>
        <Grid container>
          {Object.entries(userSkills).map((keyValuePair, idx) => (
            <TagsSection key={idx} keyValuePair={keyValuePair} />
          ))}
        </Grid>
      </Section>
      <StyledDivider yMargin="2rem" />
      <Section>
        <SectionTitle>About this Website</SectionTitle>
        <P>
          I built this website with{" "}
          <A href="https://www.gatsbyjs.com/">Gatsby</A> and{" "}
          <A href="https://github.com/Vagr9K/gatsby-advanced-starter">
            Gatsby Advanced Starter
          </A>
          . If you want to know more about my configuration, you can visit the{" "}
          <B>
            <A href="https://github.com/YSKuo/personal-site-v3">Github repo</A>.
          </B>
        </P>
      </Section>
    </Container>
  );
}

export default About;

const Tags = styled(Grid)`
  padding-right: 0rem;
  margin-bottom: 2rem;

  ${mediaQueryBreakpoint("sm")} {
    padding-right: 1rem;
  }
`;

const TagContainer = styled(Grid)`
  padding: 0.25rem;
`;

const DummyButton = styled(Button)`
  text-transform: none;
`;

const Tag = ({ tag }) => {
  return (
    <TagContainer item>
      <DummyButton key={tag} variant="contained">
        {tag}
      </DummyButton>
    </TagContainer>
  );
};

const TagsSection = ({ keyValuePair }) => {
  const [type, list] = keyValuePair;
  let category = `${type.slice(0, 1).toUpperCase()}${type.slice(1)}`; // capitalize
  switch (type) {
    case "programmingLan":
      category = "Programming Languages";
      break;
    case "backend":
    case "frontend":
      category = `${category.slice(0, -3)}-${category.slice(-3)}`;
      break;
    default:
      break;
  }

  return (
    <Grid item xs={12} sm={6}>
      <SectionSubtitle>{category}</SectionSubtitle>
      <Tags container>
        {list.map((skill, idx) => (
          <Tag key={idx} tag={skill} />
        ))}
      </Tags>
    </Grid>
  );
};
