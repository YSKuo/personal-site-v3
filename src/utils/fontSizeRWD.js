import { fontSizes } from "../constants/font";
import { mediaQueryBreakpoint } from "../constants/breakpoint";

function fontSizeRWD(text, [xs, sm, md]) {
  return `${text} {
    ${mediaQueryBreakpoint("xs")} {
      font-size: ${fontSizes[xs]};
    }
    ${mediaQueryBreakpoint("sm")} {
      font-size: ${fontSizes[sm]};
    }
    ${mediaQueryBreakpoint("md")} {
      font-size: ${fontSizes[md]};
    }
  }`;
}

export default fontSizeRWD;
