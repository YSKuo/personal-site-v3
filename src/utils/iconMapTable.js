import React from "react";

import InstagramIcon from "@material-ui/icons/Instagram";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import GitHubIcon from "@material-ui/icons/GitHub";
import EmailIcon from "@material-ui/icons/Email";
import TwitterIcon from "@material-ui/icons/Twitter";

const iconMapTable = {
  Email: <EmailIcon />,
  Twitter: <TwitterIcon />,
  GitHub: <GitHubIcon />,
  Instagram: <InstagramIcon />,
  LinkedIn: <LinkedInIcon />,
};

export default iconMapTable;
