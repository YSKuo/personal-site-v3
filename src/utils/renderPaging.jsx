import React from "react";
import styled from "styled-components";
import { Grid, Button, IconButton, Hidden } from "@material-ui/core";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";

function renderPaging(pageName, pageContext, location) {
  const { currentPageNum, pageCount } = pageContext;
  const prevPage =
    currentPageNum - 1 === 1
      ? `/${pageName}`
      : `/${pageName}/${currentPageNum - 1}/`;
  const nextPage = `/${pageName}/${currentPageNum + 1}/`;
  const isFirstPage = currentPageNum === 1;
  const isLastPage = currentPageNum === pageCount;
  const currentPage = location.pathname.match(/[0-9]+/)
    ? Number(location.pathname.match(/[0-9]+/))
    : 1;

  return (
    <Grid container component="nav" alignItems="center">
      <PaginationPreviousButton item xs={6} sm={2}>
        {!isFirstPage && (
          <Button
            href={prevPage}
            color="primary"
            variant="outlined"
            startIcon={<ArrowBackIosIcon />}
          >
            Previous
          </Button>
        )}
      </PaginationPreviousButton>

      <Hidden xsDown>
        <Grid item sm={8}>
          <PaginationList>
            {[...Array(pageCount)].map((_val, index) => {
              const pageNum = index + 1;
              if (
                pageNum === 1 ||
                pageNum === pageCount ||
                pageNum === currentPage ||
                pageNum === currentPage + 1 ||
                pageNum === currentPage - 1
              ) {
                return (
                  <PaginationNumber key={`${pageName}-page-${pageNum}`}>
                    <PaginationLink
                      variant={pageNum === currentPage ? "contained" : "text"}
                      size={pageNum === currentPage ? "medium" : "small"}
                      color={pageNum === currentPage ? "secondary" : "default"}
                      href={
                        pageNum === 1
                          ? `/${pageName}`
                          : `/${pageName}/${pageNum}/`
                      }
                      disableElevation
                    >
                      {pageNum}
                    </PaginationLink>
                  </PaginationNumber>
                );
              }
              if (pageNum === currentPage + 2 || pageNum === currentPage - 2) {
                return (
                  <PaginationNumber key={`${pageName}-page-${pageNum}`}>
                    <IconButton title="ellipsis" disabled>
                      <MoreHorizIcon />
                    </IconButton>
                  </PaginationNumber>
                );
              }
            })}
          </PaginationList>
        </Grid>
      </Hidden>
      <PaginationNextButton item xs={6} sm={2}>
        {!isLastPage && (
          <Button
            href={nextPage}
            color="primary"
            variant="outlined"
            endIcon={<ArrowForwardIosIcon />}
          >
            Next
          </Button>
        )}
      </PaginationNextButton>
    </Grid>
  );
}

export default renderPaging;

const PaginationList = styled.ul`
  display: flex;
  text-align: center;
  justify-content: center;
  align-items: center;
  list-style: none;
`;
const PaginationNumber = styled.li``;
const PaginationLink = styled(Button)`
  height: 2rem;
  width: 2rem;
  border-radius: 50%;
  min-width: initial;
  padding: 0 0.5rem;
  margin: 0 1rem;
`;

const PaginationPreviousButton = styled(Grid)`
  text-align: start;
`;

const PaginationNextButton = styled(Grid)`
  text-align: end;
`;
