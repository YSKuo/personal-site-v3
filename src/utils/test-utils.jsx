import React from "react";
import { ThemeProvider, createGlobalStyle } from "styled-components";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import { render } from "@testing-library/react";
import { theme } from "../constants/theme";

const SizeWrapper = ({ children }) => {
  const initTheme = createMuiTheme({
    props: { MuiWithWidth: { initialWidth: "lg" } },
  });

  return (
    <MuiThemeProvider theme={initTheme}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </MuiThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: SizeWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, SizeWrapper };
