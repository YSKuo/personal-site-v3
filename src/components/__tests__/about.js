import React from "react";
import userEvent from "@testing-library/user-event";

import { render } from "../../utils/test-utils";
import AboutPage from "../Pages/AboutPage";
import config from "../../../data/SiteConfig";

function renderAboutPage(props) {
  const utils = render(
    <main>
      <AboutPage config={config} />
    </main>
  );

  const about = utils.getByRole("main");
  return { ...utils, about };
}

test("AboutPage snapshot at lg screen width", () => {
  const { about } = renderAboutPage();
  expect(about).toMatchSnapshot();
});
