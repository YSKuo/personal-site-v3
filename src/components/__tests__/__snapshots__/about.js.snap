// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage snapshot at lg screen width 1`] = `
<main>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <section
      class="sc-gsTCUz cqEcIX"
    >
      <h2
        class="MuiTypography-root sc-bdfBwQ kKxBbl MuiTypography-h2 MuiTypography-gutterBottom"
      >
        About Me
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Hi, there! I'm 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Yan-Sheng (Arsene) Kuo
        </b>
        , a frontend developer and designer living in 
        Taipei, Taiwan
        . After finishing my master's degree in mechanical engineering, I acquired extensive experience in a patent firm and also the high-tech manufacturing industry.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Gaining a glimpse of user-centered methodologies from my experience in the manufacturing industry, I am eager to encourage interactions among people through digital products. Therefore, I've worked hard to improve my coding and design skills, which allow me to create websites that are intuitive and aesthetically balanced.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        In my spare time, I like to indulge myself in 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          music
        </b>
        , broaden my horizons with 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          travel
        </b>
        , document my journey by 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          photographs
        </b>
        , cultivate my taste in 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          denim
        </b>
         and 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          heritage menswear
        </b>
        , and feed my imagination of impossibilities with 
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          video games
        </b>
         and
         
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          films
        </b>
        .
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        You may also know more about me through my
         
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://www.linkedin.com/in/yanshengkuo/"
            target="_blank"
          >
            LinkedIn
          </a>
        </b>
         
        or
         
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://www.instagram.com/arsene_alibi/"
            target="_blank"
          >
            Instagram
          </a>
        </b>
        .
      </p>
    </section>
    <hr
      class="MuiDivider-root sc-hKgILt hCVDlJ"
      ymargin="2rem"
    />
    <section
      class="sc-gsTCUz cqEcIX"
    >
      <h2
        class="MuiTypography-root sc-bdfBwQ kKxBbl MuiTypography-h2 MuiTypography-gutterBottom"
      >
        Skills
      </h2>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Development
          </h3>
          <div
            class="MuiGrid-root sc-eCssSg fpZtJM MuiGrid-container"
          >
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  JavaScript (ES6+)
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  HTML
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  CSS
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  PHP
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Python
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Git
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Frontend
          </h3>
          <div
            class="MuiGrid-root sc-eCssSg fpZtJM MuiGrid-container"
          >
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  React.js
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Redux
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Gatsby.js
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  jQuery
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Webpack
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Babel
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Gulp
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Backend
          </h3>
          <div
            class="MuiGrid-root sc-eCssSg fpZtJM MuiGrid-container"
          >
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Node.js
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Django
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  MySQL
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  GraphQL
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Design
          </h3>
          <div
            class="MuiGrid-root sc-eCssSg fpZtJM MuiGrid-container"
          >
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Sketch
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Photoshop
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Figma
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Zeplin
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Wireframing
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Prototyping
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            General
          </h3>
          <div
            class="MuiGrid-root sc-eCssSg fpZtJM MuiGrid-container"
          >
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Web Application
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  RWD
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Interaction Design
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Scrum
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  SEO
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root sc-jSgupP lnWsvc MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gKsewC ktDEMV"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Unit Testing
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
    <hr
      class="MuiDivider-root sc-hKgILt hCVDlJ"
      ymargin="2rem"
    />
    <section
      class="sc-gsTCUz cqEcIX"
    >
      <h2
        class="MuiTypography-root sc-bdfBwQ kKxBbl MuiTypography-h2 MuiTypography-gutterBottom"
      >
        About this Website
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        I built this website with
         
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.gatsbyjs.com/"
          target="_blank"
        >
          Gatsby
        </a>
         and
         
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://github.com/Vagr9K/gatsby-advanced-starter"
          target="_blank"
        >
          Gatsby Advanced Starter
        </a>
        . If you want to know more about my configuration, you can visit the
         
        <b
          class="MuiTypography-root sc-dlfnbm koWLPh MuiTypography-body1 MuiTypography-gutterBottom"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://github.com/YSKuo/personal-site-v3"
            target="_blank"
          >
            Github repo
          </a>
          .
        </b>
      </p>
    </section>
  </div>
</main>
`;
